cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -446,11 +446,18 @@ def get_user_specific_instrument_list(app):
                 email = tokenHelper.get_token_user_email_address(decoded_token)
 
         out_instrument_list = []
-        for instrument_factory in importer.instrument_factory_list:
-            instrument = instrument_factory()
+        for instrument_factory in importer.instrument_factory_iter:
+            if hasattr(instrument_factory, 'instrument_query'):
+                instrument_query = instrument_factory.instrument_query
+                instr_name = getattr(instrument_factory, 'instr_name', instrument_factory().name)
+            else:
+                instrument = instrument_factory()
+                instrument_query = instrument.instrumet_query
```

Comment:
The `n` is still occasionally missing, but not everywhere.

```suggestion
                instrument_query = instrument.instrument_query
```

Reply:
That's because it's missed in `Instrument` class. This should be eventually changed, but let's do it separately. It also affects all the plugins.
I already replied to @burnout87 https://github.com/oda-hub/dispatcher-app/pull/661#discussion_r1502361312